django_project/changes/tests/test_views.py

```diff
@@ -752,6 +752,349 @@ def test_VersionDelete_no_login(self):
+        self.client = Client()
+        self.client.post(
+            '/set_language/', data={'language': 'en'})
+        logging.disable(logging.CRITICAL)
+        self.project = ProjectF.create(name='testproject')
+        self.version = VersionF.create(
+            project=self.project,
+            name='1.0.1')
+        self.category = CategoryF.create(
+            project=self.project,
+            name='testcategory')
+
+        self.user = UserF.create(**{
+            'username': 'sonlinux',
+            'password': 'password',
+        })
+        # Something changed in the way factoryboy works with django 1.8
+        # I think - we need to explicitly set the users password
+        # because the core.model_factories.UserF._prepare method
+        # which sets the password is never called. Next two lines are
+        # a work around for that - sett #581
+        self.user.set_password('password')
+        self.user.save()
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def tearDown(self):
+        """
+        Teardown after each test.
+
+        :return:
+        """
+        self.project.delete()
+        self.version.delete()
+        self.category.delete()
+        self.user.delete()
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionListView_with_normal_user(self):
+        """
+        Test if a normal user can view a list of version entries.
+        """
+        self.client.login(username='sonlinux', password='password')
+        response = self.client.get(reverse('version-list', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        expected_templates = [
+            'version/list.html', u'changes/version_list.html'
+        ]
+
+        self.assertEqual(
+            response.status_code, 200)
+        self.assertEqual(
+            response.template_name, expected_templates)
+        self.assertEqual(
+            response.context_data['object_list'][0],
+                         self.version)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionCreateView_with_normal_user(self):
+        """
+        Test if a normal user can create a list of version entries.
+        """
+        self.client.login(username='sonlinux', password='password')
+        response = self.client.get(reverse('version-create', kwargs={
+            'project_slug': self.project.slug
+        }))
+
+        self.assertEqual(
+            response.status_code, 200)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionUpdateView_with_normal_user(self):
+        """
+        Test if normal user can update a version entry.
+        """
+        self.client.login(username='sonlinux', password='password')
+        response = self.client.get(reverse('version-update', kwargs={
+            'project_slug': self.project.slug,
+            'slug': self.version.slug
+        }))
+        self.assertEqual(
+            response.status_code, 404)
```

Comment:
@timlinux Can you confirm that a normal user (neither staff nor changelog manager) can't update a version?

If so @sonlinux please go and fix the HTML template if the same pattern as you did in this PR. I'm loggedin with a normal account on http://changelog.qgis.org/en/qgis/version/3.2.0/ I can see the button to edit the version and then I get the 404 you described here. This is not correct.